content/2020/08-28-multi-select-metadata-support-in-search-api.md

```diff
@@ -0,0 +1,54 @@
+---
+applied_at: "2020-08-28"
+applies_to: 
+  - api
+is_impactful: false
+is_new_feature: true
+collapse: true
+show_excerpt: true
+release_source_url: ''
+---
+
+# New `multiSelect` metadata support in search API
+
+Starting today, the [Search API][search] adds support for searching for items
+by multiple values of a [`multiSelect` metadata field][multi_select]. Before this
+change, it was not possible to find items that matched multiple values for a
+`multiSelect` field.
+
+To perform a search for items where a metadata field matches
+multiple values, the `mdfilters` parameter now supports a list of values.
+
+```curl
+curl -G 'https://api.box.com/2.0/search' \
+     -H 'Authorization: Bearer <ACCESS_TOKEN>' \
+     -d 'mdfilters=[{"scope":"enterprise_12345","templateKey":"contractInfo","filters":{"products":["shield","platform"]}}]'
+```
+
+<!-- more -->
+
+In this example, the `mdfilters` query parameter contains one filter with a
+`scope`, `templateKey`, a set of `filters`. Here is the same filter in a more
+readable format.
+
+```json
+[
+  {
+    "scope": "enterprise_12345",
+    "templateKey": "contractInfo",
+    "filters": {
+      "products": [
+        "shield",
+        "platform"
+      ]
+    }
+  }
+]
+```
+
+What is new here is that the `products` filter now performs a match on multiple
+values, only returning files and folders for which the template has a `products`
+value of both `shield` and `platform`.
```

Comment:
I've updated the language but would like to clarify something.

Assume I have a template with a `multiSelect` field name `products` and I have an instance for that template that has the values `shield,platform`.

Now, if I perform a search for `products=shield,governance`, will this instance match (`shield` is in the list of values afterall) or will it not match (because the value also includes `governance`)?

Reply:
This instance will match because it includes `shield`